### PR TITLE
Add mapper 2.1 functions support

### PR DESCRIPTION
**Pull request by yhe39**
Change-Id: I1a27625e2dc14cae5957936dbae7c9111ed8a07b
Signed-off-by: HeYue <yue.he@intel.com>

### CODE REVIEW

**Comment by yhe39** [comment]
@js0701 @YuanjunHuang , could u help to review?
i include the johnson's patch to delete handles_ in gralloc driver.
modify the import buffer function, use the clone buffer then retain buffer to import buffer.
delete the format check in validate buffer size.
now with the two framework patch and this patch, the system could boot to UI successfully.
Thanks!

**Comment by js0701** [comment]
@yhe39  Are all those steps are a must necessary for successful boot?
How about the VTS test cases

**Comment by yhe39** [comment]
The two:
"modify the import buffer function, use the clone buffer then retain buffer to import buffer.
delete the format check in validate buffer size."
are must to successfully boot. 
not sure whether your change "patch to delete handles_ in gralloc driver" is necessary.
i will check.
will check the vts test too.
Thanks!

**Comment by yhe39** [comment]
hi @js0701 @YuanjunHuang , could u help to review. The vts mapper2.1 related test all pass.
Thanks!

File: cros_gralloc/gralloc1/cros_gralloc1_module.cc [hunk 1]
```diff
@@ -299,12 +305,75 @@ int32_t CrosGralloc1::setInterlace(buffer_handle_t buffer, uint32_t interlace)
...
+	}
+
+	// Do not support GRALLOC1_CAPABILITY_LAYERED_BUFFERS, only allocate buffers with a
+	// single layer.
+	if (descriptorInfo->layerCount != 1) {
+		return CROS_GRALLOC_ERROR_BAD_VALUE;
+	}
+	if (stride > hnd->pixel_stride || descriptorInfo->width > hnd->width ||
+	    descriptorInfo->height > hnd->height) {
+
```
Comment (YuanjunHuang) [inline cros_gralloc/gralloc1/cros_gralloc1_module.cc]: remove blank line.

**Comment by js0701** [comment]
LGTM. Thanks

File: cros_gralloc/gralloc1/cros_gralloc1_module.cc [hunk 2]
```diff
@@ -299,12 +305,75 @@ int32_t CrosGralloc1::setInterlace(buffer_handle_t buffer, uint32_t interlace)
...
+int32_t CrosGralloc1::validateBufferSize(buffer_handle_t buffer,
+					 const gralloc1_buffer_descriptor_info_t *descriptorInfo,
+					 uint32_t stride)
+{
+	auto hnd = (cros_gralloc_handle *)cros_gralloc_convert_handle(buffer);
+	if (!hnd) {
+		return CROS_GRALLOC_ERROR_BAD_HANDLE;
+	}
+
+	if (cros_gralloc_convert_format(descriptorInfo->format) != hnd->format) {
```
Comment (YuanjunHuang) [inline cros_gralloc/gralloc1/cros_gralloc1_module.cc]: the api intends to validate buffer size, why here validates buffer format? is it required to be tested/checked by test case?

Comment (yhe39) [reply cros_gralloc/gralloc1/cros_gralloc1_module.cc]: The vts test case check the height, width, stride, format and layerCount. :)
Not requried from the api description.

Comment (YuanjunHuang) [reply cros_gralloc/gralloc1/cros_gralloc1_module.cc]: ok. thanks. LGTM.

**Comment by yhe39** [comment]
Thanks!

**Comment by yhe39** [comment]
close as merged

**Pull request closed without merging**